### PR TITLE
refactor(format): enable get_splits() to be used independently.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

so we can use a split as a table partition later.

Closes #issue

### CODE REVIEW

**Review by BohuTANG — APPROVED** [review]

File: src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs [hunk 1]
```diff
@@ -340,6 +346,8 @@ impl AligningStateTrait for AligningState {
         if let Some(rb) = read_batch {
             if let ReadBatch::Buffer(b) = rb {
                 self.buffers.push(b)
+            } else {
+                unreachable!()
```
Comment (BohuTANG) [inline src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs:350]: Return error?

Comment (youngsofun) [reply src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs:350]: is `ErrorCode::Internal` suitable for this?

Comment (BohuTANG) [reply src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs:350]: Ok

Comment (youngsofun) [reply src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_parquet.rs]: done

**Comment by BohuTANG** [comment]
@mergify update

**Pull request merged**